Core/Core/Grades/View/GradeRowView.swift

```diff
@@ -65,7 +65,11 @@ public struct GradeRowView: View {
             let submission = assignment.submissions?.first { $0.userID == userID }
             let status = submission?.status ?? .notSubmitted
 
-            Text(status.text)
+            let statusText = submission?.status == .submitted && submission?.needsGrading == false
```

Comment:
Even though this logic is rather simple, but it's used in multiple places, would it make sense to extract it to a new computed property in Submission?
It's a bit confusing there that there is `needsGrading` and `isGraded` already, I think it would help a bit to not add a third kind of logic elsewhere.

Reply:
They (`needsGrading`, `isGraded`) all serve different purposes, so a new, third property will contain this logic anyway. But it's a good idea, so I'll extract it.